File: .ProgramingBasics/DrawingFiguresWithLoops/05.SquareFrame/src/SquareFrame.java
import java.util.Scanner;

public class SquareFrame {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());

        System.out.println("+" + repeatStr(" -", n - 2) + " +");

        for (int i = 0; i < n - 2; i++) {
            System.out.println("|" + repeatStr(" -", n - 2) + " |");
        }

        System.out.println("+" + repeatStr(" -", n - 2) + " +");
    }

    private static String repeatStr(String str, int count) {
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < count; i++) {
            sb.append(str);
        }

        return sb.toString();
    }
}

File: ProgramingBasics/DrawingFiguresWithLoops/05.SquareFrame/05.SquareFrame/SquareFrame.cs
using System;
using System.Text;

namespace _05.SquareFrame
{
    public class SquareFrame
    {
        public static void Main(string[] args)
        {
            var n = int.Parse(Console.ReadLine());

            Console.WriteLine($"+ {RepeatStr(" -", n - 2)} +");

            for (var i = 0; i < n - 2; i++)
            {
                Console.WriteLine("|" + RepeatStr(" -", n - 2) + " |");
            }

            Console.WriteLine("+" + RepeatStr(" -", n - 2) + " +");
        }

        private static string RepeatStr(string str, int count)
        {
            var sb = new StringBuilder();

            for (var i = 0; i < count; i++)
            {
                sb.Append(str);
            }

            return sb.ToString();
        }
    }
}
